test/parallel/test-internal-module-wrap.js

```diff
@@ -22,4 +22,28 @@ const bar = new ModuleWrap('bar', undefined, 'export const five = 5', 0, 0);
 
   // Check that the module requests are the same after linking, instantiate, and evaluation.
   assert.deepStrictEqual(moduleRequests, foo.getModuleRequests());
+}
+
+// Verify that linking two module with a same ModuleCacheKey throws an error.
+function testLinkMismatch() {
+  const foo = new ModuleWrap('foo', undefined, `
+    import source BarSource from 'bar';
+    import bar from 'bar';
+`, 0, 0);
+  const bar1 = new ModuleWrap('bar', undefined, 'export const five = 5', 0, 0);
+  const bar2 = new ModuleWrap('bar', undefined, 'export const six = 6', 0, 0);
+
+  const moduleRequests = foo.getModuleRequests();
+  assert.strictEqual(moduleRequests.length, 2);
```

Comment:
This is surprising to me - I'd have expected requests to be deduplicated on their unique identity, such that this would be a single request.